tests/utils.go

```diff
@@ -4918,3 +4923,7 @@ func DetectLatestUpstreamOfficialTag() (string, error) {
 	By(fmt.Sprintf("By detecting latest upstream official tag %s for current branch", tag))
 	return tag, nil
 }
+
+func retcode(retcode string) string {
+	return "\n" + retcode
```

Comment:
Can you please explain why the "\n" is needed here?

Reply:
without it , if you write a command 
ip route add .... f10:....
and then do echo $?
and EXP for 0
the 0 of the command would enter the buffer, and it will match for the EXP

you can see the buffer and the matching if you use verbose or better imo by dumping the structs which say exactly which EXP it correspond to.

with \n , since we are not in multiline mode of re2, it will enforce that its on a new line and not the output of the command that we used (unless we have HEREDOC or something ,which we dont)

Reply:
Sorry, I still don't understand.
Isn't it a good think the 0 return by echo $?\n will match the Exp?

Reply:
its not returned by the echo $? 
its the 0 that is part of the SND command itself
since its in the buffer that EXP look on

you can se here when it say {11 ... what it matches
```
----

the good

{"component":"tests","kind":"VirtualMachineInstance","level":"info","msg":"Configure ipv6 GOOD: [{1 \r\n$  [$ ]} {3 \r\n$  [$ ]} {5 sudo ip -6 addr add fd10:0:2::2/120 dev eth0\r\n$ echo $?\r\n0 [\n0]} {7 \r\n$  [$ ]} {9 sleep 5\r\n$  [$ ]} {11 sudo ip -6 route add default via fd10:0:2::1 src fd10:0:2::2\r\n$ echo $?\r\n0 [\n0]}]","name":"testvmi2r97k92hd8mjxsdp84n4sszdhw65tzz2wn79krfrjwwk9vzz","namespace":"kubevirt-test-default","pos":"utils.go:2917","timestamp":"2020-06-15T06:59:56.111569Z","uid":"05a36d8f-91e4-4fa3-9ed6-52a432a2a626"}


the bad

{"component":"tests","kind":"VirtualMachineInstance","level":"info","msg":"Configure ipv6 GOOD: [{1 \r\n$  [$ ]} {3 \r\n$  [$ ]} {5 sudo ip -6 addr add fd10 [0]} {7 :0:2::2/120 dev eth0\r\n$  [$ ]} {9 echo $?\r\n0\r\n$  [$ ]} {11 sleep 5\r\n$ sudo ip -6 route add default via fd10 [0]}]","name":"testvmi86vtkr99wdp7mlz7t7jjc4gk5jrds9z924n2l4p4mqzm58fr","namespace":"kubevirt-test-default","pos":"utils.go:2917","timestamp":"2020-06-15T07:01:26.190329Z","uid":"8c931971-b5db-4e11-9653-8996054547e6"}

---
```

Reply:
oh, it considers the 0 in the ip address as the EXP! Now I understand. Thanks for the explanation. Can you please change the PR message so it will be more clear.

Reply:
updated, hope its better now

Reply:
It's better but it is still unclear the EXP caught the '0' character of the ip address.